File: python/armstrong_numbers.py
# How can you make this more scalable and reusable later?
import math

def find_armstrong_numbers(numbers):
    result = [];
    for number in numbers:
        expo = len(str(number))
        test_num = [int(n) for n in str(number)]
        sum = math.floor(math.fsum(n**expo for n in test_num))
        if sum == number:
            result.append(sum)
    return result

#print(find_armstrong_numbers(list(range(0,15))))

File: js/armstrongNumbers.js
// How can you make this more scalable and reusable later?

exports.findArmstrongNumbers = function(input) {
//let findArmstrongNumbers = function(input) {

    let result = [];
    let numString;
    let answer;

    for (let num = 0; num < input; num++) {
        answer = 0;
        numString = num.toString().split('');
        expo = numString.length;
        for (let i = 0; i < numString.length; i++) {
            numString[i] = (numString[i] * 1) ** expo;
            answer += numString[i];
        }
        if (answer === num) {
            result.push(answer)
        }
    }

    return result

};

//findArmStrongNumbers()

//console.log(findArmstrongNumbers(235))
